src/clj/rems/workflow/dynamic.clj

```diff
@@ -196,8 +198,150 @@
                              :event/actor "foo"
                              :application/id 123}))))))
 
+;;; Roles and permissions
 
-;;; Events
+(defn hide-sensitive-dynamic-events [events]
+  (filter (fn [event]
+            ((complement contains?) #{:application.event/comment-requested
+                                      :application.event/commented
+                                      :application.event/decided
+                                      :application.event/decision-requested}
+             (:event/type event)))
+          events))
+
+(defmulti calculate-permissions
```

Comment:
I feel like this function is not working so well because it doesn't actually calculate the permissions. You have to have a separate step to remember to remove "impossible" permissions.

Reply:
The impossible permissions filtering is there just for backward compatibility with the old implementation, so that the list of possible commands would be the same. It's not so much about permissions, but about application state: you're allowed to remove members, but if there are no members then you can't remove anything, even though technically you still have full authority to remove members. Assuming the UI has a remove member button next to each member, then when there are no members, the button is never rendered regardless of whether the user is authorized to remove members or not. The commands also anyways validate when the user tries to remove a member that the member exists. Once the the old permission system based on `impossible-command?` is removed, then `remove-impossible-permissions` can also be removed without affecting functionality or security.

Reply:
If that becomes possible and all permissions are updated by the events in `calculate-permissions` then it's good.